src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -149,7 +146,7 @@ internal override Task<RazorCodeDocument> GetLatestCodeDocumentAsync(ITextSnapsh
                     _codeDocumentRequests.Add(request);
                 }
 
-                return request.Task;
+                return request.Task!;
```

Comment:
More scary. Returning a null task would cause all sorts of nastiness.

Reply:
This suppression is required to convert `Task<RazorCodeDocument>` -> `Task<RazorCodeDocument?>`. Yes, is is ugly and scary unless you dig into why it is made, but how can it actually be done better?

Reply:
Investigated this one a little bit. First, compiler doesn't allow the conversion to be implicit, even though we convert non-nullable generic parameter into nullable, which is safe, sin't it? Second, the `Task` property comes from `TaskCompletionSource` in the end, and making it null-annotated will just move suppression to the other part of the code. At the same time the result of this method is 100% nullable, there is even the null check in at least one of its usages. Here is a gif:
![devenv_XXZYxypDyN](https://user-images.githubusercontent.com/70431552/193525796-7a25c0dc-cbff-4d01-bb5e-73ae9bc77140.gif)

Reply:
Maybe just add an explanatory comment and be done with that?

Reply:
Sorry, meant to look at this today, but missed it. Will pull this branch tomorrow and have a look.

Reply:
Okay, looked at this, and yes, I think an explanitory comment would be perfect. It looks like its suppressing such that it could be returning `Task<RazorCodeDocument>?`, which is scary, but suppressing because there is no conversion from `Task<RazorCodeDocument>` to `Task<RazorCodeDocument?>` seems perfectly fine.